### PR TITLE
add in keyfile support

### PR DESCRIPTION
**Pull request by addisonj**
This is a cherry-picked clean commit from #40

This adds key-file support

Credit goes to @lackac

### CODE REVIEW

**Comment by dcrosta** [comment]
We need to also adjust the "defaults" template for other platforms. Otherwise LGTM.

File: recipes/default.rb [hunk 1]
```diff
@@ -33,6 +33,17 @@
   Gem.clear_paths
 end
 
+# Create keyFile if specified
+if node[:mongodb][:key_file]
+  file "/etc/mongodb.key" do
+    owner "mongodb"
+    group "mongodb"
```
Comment (dcrosta) [inline recipes/default.rb]: these should be `node[:mongodb][:user]` and `node[:mongodb][:group]`, respectively

File: templates/freebsd/mongodb.default.erb [hunk 2]
```diff
@@ -17,6 +17,10 @@ DAEMON_OPTS="--fork"
 DAEMON_OPTS="$DAEMON_OPTS --replSet <%= @replicaset_name %>"
 <% end %>
 
+<% if node[:mongodb][:key_file] %>
+DAEMON_OPTS="$DAEMON_OPTS --keyFile /etc/mongodb.key"
```
Comment (addisonj) [inline templates/freebsd/mongodb.default.erb:21]: Should the key file path be configurable via an attribute?

Comment (dcrosta) [inline templates/freebsd/mongodb.default.erb:21]: We can always add that later. In most cases, any given machine won't be part of two separate clusters with different keyFiles, in my experience.

**Comment by dcrosta** [comment]
Any other feedback, maintainers?

**Comment by miketheman** [comment]
One nitpick - attribute name `key_file` vs parameter `keyFile`. Nothing world-breaking, just if we choose a convention for converting camelCase parameters to under_scores, then we do it all the way.

**Comment by vitalis** [comment]
+1 
really need this feature

**Comment by apicht-spireon** [comment]
+1

**Comment by 0xAhmed** [comment]
+1

**Comment by rricard** [comment]
+1

**Pull request merged**